packages/react-dom/src/__tests__/quoteAttributeValueForBrowser-test.js

```diff
@@ -9,40 +9,64 @@
 
 'use strict';
 
+var ExecutionEnvironment;
+var React;
+var ReactDOMServer;
+
+var ROOT_ATTRIBUTE_NAME;
+
 describe('quoteAttributeValueForBrowser', () => {
-  // TODO: can we express this test with only public API?
-  var quoteAttributeValueForBrowser = require('../shared/quoteAttributeValueForBrowser')
-    .default;
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+
+    ExecutionEnvironment = require('fbjs/lib/ExecutionEnvironment');
+    ExecutionEnvironment.canUseDOM = false;
+    ReactDOMServer = require('react-dom/server');
 
-  it('should escape boolean to string', () => {
-    expect(quoteAttributeValueForBrowser(true)).toBe('"true"');
-    expect(quoteAttributeValueForBrowser(false)).toBe('"false"');
+    // TODO: can we express this test with only public API?
+    var DOMProperty = require('../shared/DOMProperty');
+    ROOT_ATTRIBUTE_NAME = DOMProperty.ROOT_ATTRIBUTE_NAME;
   });
 
-  it('should escape object to string', () => {
-    var escaped = quoteAttributeValueForBrowser({
-      toString: function() {
-        return 'ponys';
-      },
-    });
+  it('ampersand is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr="&" />);
+    expect(response).toMatch(
+      new RegExp(
+        '<img data-attr="&amp;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>',
+      ),
+    );
+  });
 
-    expect(escaped).toBe('"ponys"');
+  it('double quote is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr={'"'} />);
+    expect(response).toMatch(
+      new RegExp(
+        '<img data-attr="&quot;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>',
+      ),
+    );
   });
 
-  it('should escape number to string', () => {
-    expect(quoteAttributeValueForBrowser(42)).toBe('"42"');
+  it('single quote is escaped inside attributes', () => {
+    var response = ReactDOMServer.renderToString(<img data-attr="'" />);
+    expect(response).toMatch(
+      new RegExp(
+        '<img data-attr="&#x27;" ' + ROOT_ATTRIBUTE_NAME + '=""' + '/>',
+      ),
+    );
   });
 
-  it('should escape string', () => {
-    var escaped = quoteAttributeValueForBrowser(
-      '<script type=\'\' src=""></script>',
```

Comment:
Please add a test where this is literally the argument value.

Reply:
@gaearon I'm thinking of creating tests for escapeText thourgh the ReactDOMServer.renderToString API like these ones, do you think that it would make sense to write them? Also, if you do, shouldn't this test be placed there where the script would actually run?

Reply:
>I'm thinking of creating tests for escapeText thourgh the ReactDOMServer.renderToString API like these ones

Sounds goood.

>Also, if you do, shouldn't this test be placed there where the script would actually run?

Not sure what you mean. We just want to verify `<script>` can't be injected as attribute value.

Reply:
Great, will add test on both escapeText and quoteAttribute tests.